lib/Spot/Query.php

```diff
@@ -651,32 +666,63 @@ public function escape($string)
     }
 
     /**
-     * Escape/quote direct user input
+     * Escape/quote multiple identifiers
      *
-     * @param string $field
+     * @param array $identifiers
+     * @return array
+     */
+    protected function escapeMultipleIdentifiers(array $identifiers) {
+        array_walk($identifiers, function (&$identifier) {
+            $identifier = $this->escapeIdentifier($identifier);
+        });
+
+        return $identifiers;
+    }
+
+    /**
+     * Removes escape/quote character
+     *
+     * @param string $identifier
      * @return string
-     * @throws Exception
      */
-    public function escapeField($field)
+    public function unescapeIdentifier($identifier)
     {
-        if ($this->_noQuote) {
-            return $field;
+        if (strpos($identifier, ".") !== false) {
+            $parts = array_map(array($this, "unescapeIdentifier"), explode(".", $identifier));
+
+            return implode(".", $parts);
         }
 
-        return $this->mapper()->connection()->quoteIdentifier($field);
+        return trim($identifier, $this->mapper()->connection()->getDatabasePlatform()->getIdentifierQuoteCharacter());
+    }
+
+    /**
+     * Escape/quote identifier
+     *
+     * @param string $identifier
+     * @return string
+     */
+    public function escapeIdentifier($identifier)
```

Comment:
Remove the 'escapeMultipleIdentifiers` method above, and just cast `$identifier` to an array in this function so it can handle both strings and arrays:

```php
foreach((array) $identifiers as $identifier) {
// ...
```